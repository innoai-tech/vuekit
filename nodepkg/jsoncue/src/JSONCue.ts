import {
  Source,
  stringify,
  toValue,
  asBytes,
  stringifyPropertyName,
} from "./astutil";

export { SyntaxError } from "./astutil";

export abstract class JSONCue {
  static parse<T extends any>(text: string): T {
    const src = Source.parse(text);
    return toValue(src, src.topNode) as unknown as T;
  }

  static stringify(v: any): string {
    return stringify(v);
  }

  static asBytes(s: string) {
    return asBytes(s);
  }

  static stringifyPropertyName(propName: string) {
    return stringifyPropertyName(propName);
  }
}
