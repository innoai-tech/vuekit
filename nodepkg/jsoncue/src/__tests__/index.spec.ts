import { describe, expect, it } from "bun:test";
import { JSONCue } from "../JSONCue.ts";

describe("JSONCue", () => {
  const simple = Bun.file(
    Bun.fileURLToPath(import.meta.resolve("./fixtures/simple.json.cue")),
  );

  it("should parse", async () => {
    const v = JSONCue.parse(await simple.text());

    expect(v).toEqual({
      str: "str value",
      mlStr: "1\n\t22\n\t\t333\n",
      bytes: JSONCue.asBytes("{}"),
      mlBytes: JSONCue.asBytes(`a:\n  b: 1\n`),
      int: 1,
      float: 1.1,
      array: [true, false, null],
      nested: {
        a: {
          b: { c: 2, d: 3 },
        },
      },
    });

    expect(JSONCue.stringify(JSON.parse(JSON.stringify(v)))).toMatchSnapshot();
  });
});
