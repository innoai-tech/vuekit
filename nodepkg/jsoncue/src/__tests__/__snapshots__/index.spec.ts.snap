// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`JSONCue should parse 1`] = `
"mlStr:        """
  1
  	22
  		333
  """
str:          "str value"
bytes:        "e30="
mlBytes:      "YToKICBiOiAxCg=="
int:          1
float:        1.1
array:        [
  true,
  false,
  null,
]
nested: a: b: {
  c: 2
  d: 3
}
"
`;

exports[`JSONCue should parse 1`] = `
"mlStr:        """
  1
  	22
  		333
  """
str:          "str value"
bytes:        "e30="
mlBytes:      "YToKICBiOiAxCg=="
int:          1
float:        1.1
array:        [
  true,
  false,
  null,
]
nested: a: b: {
  c: 2
  d: 3
}
"
`;

exports[`JSONCue should parse 1`] = `
"mlStr:        """
  1
  	22
  		333
  """
str:          "str value"
bytes:        "e30="
mlBytes:      "YToKICBiOiAxCg=="
int:          1
float:        1.1
array:        [
  true,
  false,
  null,
]
nested: a: b: {
  c: 2
  d: 3
}
"
`;

exports[`JSONCue should parse 1`] = `
"mlStr:        """
  1
  	22
  		333
  """
str:          "str value"
bytes:        "e30="
mlBytes:      "YToKICBiOiAxCg=="
int:          1
float:        1.1
array:        [
  true,
  false,
  null,
]
nested: a: b: {
  c: 2
  d: 3
}
"
`;

exports[`JSONCue should parse 1`] = `
"mlStr:        """
  1
  	22
  		333
  """
str:          "str value"
bytes:        "e30="
mlBytes:      "YToKICBiOiAxCg=="
int:          1
float:        1.1
array:        [
  true,
  false,
  null,
]
nested: a: b: {
  c: 2
  d: 3
}
"
`;

exports[`JSONCue should parse 1`] = `
"mlStr:        """
  1
  	22
  		333
  """
str:          "str value"
bytes:        "e30="
mlBytes:      "YToKICBiOiAxCg=="
int:          1
float:        1.1
array:        [
  true,
  false,
  null,
]
nested: a: b: {
  c: 2
  d: 3
}
"
`;

exports[`JSONCue should parse 1`] = `
"mlStr:        """
  1
  	22
  		333
  """
str:          "str value"
bytes:        "e30="
mlBytes:      "YToKICBiOiAxCg=="
int:          1
float:        1.1
array:        [
  true,
  false,
  null,
]
nested: a: b: {
  c: 2
  d: 3
}
"
`;

exports[`JSONCue should parse 1`] = `
"mlStr:        """
  1
  	22
  		333
  """
str:          "str value"
bytes:        "e30="
mlBytes:      "YToKICBiOiAxCg=="
int:          1
float:        1.1
array:        [
  true,
  false,
  null,
]
nested: a: b: {
  c: 2
  d: 3
}
"
`;

exports[`JSONCue should parse 1`] = `
"mlStr:        """
  1
  	22
  		333
  """
str:          "str value"
bytes:        "e30="
mlBytes:      "YToKICBiOiAxCg=="
int:          1
float:        1.1
array:        [
  true,
  false,
  null,
]
nested: a: b: {
  c: 2
  d: 3
}
"
`;
